src/main/java/com/hotels/beans/cache/CacheManager.java

```diff
@@ -72,17 +69,16 @@
      * @param <T> the class object type.
      * @return the cached object.
      */
+    @SuppressWarnings("unchecked")
     public <T> T getFromCache(final String cacheKey, final Class<? extends T> objectClass) {
-        notNull(cacheKey, "cacheKey cannot be null!");
-        notNull(objectClass, "objectClass cannot be null!");
-        return ofNullable(cacheMap.get(cacheKey)).map(objectClass::cast).orElse(null);
+        return (T) cacheMap.computeIfPresent(cacheKey, (k, v) -> objectClass.cast(v));
```

Comment:
The computeIfPresent method will set a new value for the key if the key already exists in the map or remove the key entirely if the mapping method returns null.  I don't think this is the desired result.  From the Java 11 JavaDoc:

```
If the value for the specified key is present and non-null, attempts to compute a new mapping given the key and its current mapped value.
If the remapping function returns null, the mapping is removed. If the remapping function itself throws an (unchecked) exception, the exception is rethrown, and the current mapping is left unchanged.
The remapping function should not modify this map during computation.
```

The original logic should be fine with the caveat of potentially creating a new Optional<> on every call versus handling null directly.